tests/test_gallery.py

```diff
@@ -0,0 +1,112 @@
+IMG_3870_HEIC = os.path.join(paths.TEST_DATA_DIRECTORY, "gallery/IMG_3870.heic")
+IMG_3870_JPEG = os.path.join(paths.TEST_DATA_DIRECTORY, "gallery/IMG_3870.jpeg")
+
+
+class GalleryTestCase(unittest.TestCase):
+
+    def test_regex(self):
+        self.assertTrue(Regex(r".*").evaluate("a"))
+        self.assertTrue(Regex(r"hello").evaluate("hello"))
+        self.assertFalse(Regex(r"hello").evaluate("goodbye"))
+        self.assertTrue(Regex(r"hello").evaluate("hello world"))
+        self.assertFalse(Regex(r"^hello$").evaluate("hello world"))
+
+    def test_equal(self):
+        self.assertTrue(Equal(True).evaluate(True))
+        self.assertFalse(Equal(True).evaluate(False))
+        self.assertTrue(Equal(12).evaluate(12))
+        self.assertFalse(Equal(12).evaluate(13))
+        self.assertFalse(Equal(18).evaluate("cheese"))
+        self.assertTrue(Equal("cheese").evaluate("cheese"))
+        self.assertFalse(Equal("cheese").evaluate("fromage"))
+        self.assertFalse(Equal("cheese").evaluate(12))
+
+    def test_or(self):
+        self.assertTrue(Or(Equal(True)).evaluate(True))
+        self.assertTrue(Or(Equal(True), Equal(False)).evaluate(True))
+        self.assertTrue(Or(Equal(False), Equal(True)).evaluate(True))
+        self.assertFalse(Or(Equal(False), Equal(False)).evaluate(True))
+        self.assertTrue(Or(Equal(False), Equal(False)).evaluate(False))
+
+    def test_glob(self):
+        self.assertTrue(Glob("*.jpeg").evaluate("IMG_3875.jpeg"))
+        self.assertFalse(Glob("*.tiff").evaluate("IMG_3875.jpeg"))
+        self.assertTrue(Glob("*.{jpeg,tiff}").evaluate("IMG_3875.jpeg"))
+        self.assertTrue(Glob("*.{jpeg,tiff}").evaluate("IMG_3875.tiff"))
+
+    def test_evaluate_tests(self):
+        tests = [
+            (Glob("*.tiff"), "image/jpeg"),
+            (Glob("*"), "*"),
+        ]
+        self.assertEqual(gallery.evaluate_tests(tests, "IMG_3875.jpeg"), "*")
+        self.assertEqual(gallery.evaluate_tests(tests, "IMG_3875.tiff"), "image/jpeg")
+
+    @common.with_temporary_directory
+    def test_resize_heic_with_format_change(self):
+
+        basename = os.path.basename(IMG_3870_HEIC)
+        size = (1600, None)
+        gallery.resize(IMG_3870_HEIC, os.getcwd(), "", basename, size, 1)
+
+        expected_basename = utils.replace_extension(basename, ".jpg")
+        self.assertTrue(os.path.exists(expected_basename))
+        output_size = gallery.get_size(expected_basename, 1)
+        self.assertEqual(output_size["width"], size[0])
+
+    @common.with_temporary_directory
+    def test_resize_jpeg(self):
+
+        basename = os.path.basename(IMG_3870_JPEG)
+        size = (800, None)
+        gallery.resize(IMG_3870_JPEG, os.getcwd(), "", basename, size, 1)
+
+        self.assertTrue(os.path.exists(basename))
+        output_size = gallery.get_size(basename, 1)
+        self.assertEqual(output_size["width"], size[0])
+
+        # TODO: Test TIFF and GIF conversion
```

Comment:
These tests should be fleshed out.

Reply:
Perhaps this is also the time to ensure that there's a full end-to-end test for building a site containing an image.

Reply:
These tests have now been added, and full integration tests can be added when the new image profiles are supported.